pifpaf/drivers/rabbitmq.py

```diff
@@ -50,57 +52,63 @@ def get_parser(cls, parser):
                             help="RabbitMQ password")
         return parser
 
-    def _start_node(self, nodename, port, env):
+    def start_node(self, nodename, port):
         complete_env = {
             "RABBITMQ_NODE_IP_ADDRESS": b"127.0.0.1",
             "RABBITMQ_NODE_PORT": str(port),
             "RABBITMQ_NODENAME": nodename,
             "RABBITMQ_PIDFILE": os.path.join(self.tempdir, nodename, "pid"),
         }
-        complete_env.update(env)
+        complete_env.update(self.env)
         c, _ = self._exec(["rabbitmq-server"], env=complete_env,
                           path=self._path,
                           wait_for_line=b"Starting broker... completed")
-        self.addCleanup(self._kill, c.pid)
+        self.addCleanup(self.kill_node, nodename)
+        self._process[nodename] = c
```

Comment:
Maybe add a check that there's not already nodename in the _process, ya never know.